css/src/components/site-header.scss

```diff
@@ -0,0 +1,253 @@
+
+%site-header-hover-underline {
+	> :not(.icon) {
+		position: relative;
+
+		&::after {
+			display: block;
+			position: absolute;
+			content: '';
+			border-block-end: $site-header-hover-underline-border;
+			inset-inline-start: 0;
+			inset-inline-end: 0;
+			inset-block-end: -$site-header-hover-underline-border-width;
+		}
+	}
+}
+
+%site-header-control {
+	outline-offset: -$focus-width;
+	color: $site-header-control-color;
+	text-decoration: none;
+	white-space: nowrap;
+	cursor: pointer;
+	padding-block: $button-padding-vertical;
+	padding-inline: $site-header-inline-gap;
+
+	&:visited {
+		color: $site-header-control-color;
+	}
+}
+
+.site-header {
+	display: flex;
+	position: relative;
+	align-items: center;
+	height: $site-header-height;
+	font-size: $site-header-font-size;
+	border-block-end: $site-header-border;
+	gap: $site-header-inline-gap;
+	padding-inline: $layout-gap;
+
+	@include widescreen {
+		padding-inline: $layout-widescreen-gap;
+	}
+
+	.site-header-button {
+		@include control;
+		@include unselectable;
+		@extend %site-header-control;
+
+		justify-content: center;
+		height: 100%;
+		border: none;
+		border-radius: 0;
+		background-color: transparent;
+		font-size: $site-header-button-font-size;
+		font-weight: $weight-normal;
+		text-align: center;
+
+		&[aria-expanded='true'] {
+			background-color: $site-header-panel-background-color;
+		}
+
+		.icon {
+			font-size: $site-header-button-icon-font-size;
+
+			&:only-child {
+				margin: 0;
+			}
+
+			// stylelint-disable selector-max-specificity
+
+			&:first-child:not(:only-child) {
+				margin-inline-end: $site-header-button-icon-spacing;
+			}
+
+			&:last-child:not(:only-child) {
+				margin-inline-start: $site-header-button-icon-spacing;
+			}
+			// stylelint-enable
+		}
+
+		&:active,
+		&.is-active,
+		&:hover,
+		&.is-hovered,
+		&:focus-visible {
+			&:not([aria-expanded='true']) {
+				@extend %site-header-hover-underline;
+			}
+		}
+	}
+
+	.site-header-logo {
+		display: block;
+		width: $site-header-logo-width;
+		height: auto;
+	}
+
+	.site-header-logo-centered {
+		position: absolute;
+		inset-inline-start: calc(50% - $site-header-logo-width / 2);
+	}
+
+	.site-header-divider {
+		display: inline-block;
+		height: $site-header-divider-height;
+		border-inline-start: 2px solid $text-subtle;
+		margin-inline-start: $site-header-inline-gap;
+	}
+
+	.site-header-brand {
+		@extend %site-header-control;
+
+		display: inline-block;
+		font-size: $site-header-brand-font-size-sm;
+		font-weight: $weight-semibold;
+
+		&:active,
+		&.is-active,
+		&:hover,
+		&.is-hovered,
+		&:focus-visible {
+			@extend %site-header-hover-underline;
+		}
+
+		@include tablet {
+			font-size: $site-header-brand-font-size;
+		}
+	}
+
+	.site-header-nav {
+		display: flex;
+		align-self: center;
+		width: 100%;
+		height: 100%;
+	}
+
+	.site-header-panel {
+		display: grid;
+		position: absolute;
```

Comment:
I think we can keep this as it is for now, but I think doing things like absolutely positioning components limits their usage. Panel is a decent layout component (and one that could really be even more general than it currently is), but in doing abs positioning we make using it generally a little harder.

Reply:
Good point, let me see if I can remove absolute positioning and use atomics on panels itself to achieve the same results.

Reply:
Updated! Atomics seem to be doing the thing in Learn

Reply:
Wasn't expecting a change, but since we made it, I have to ask another question. Should the panel be part of the site-header component? Right now it's nested inside, so cannot be used outside. And it's name is prefixed by site header. I think as the code stands, I'm not sure what the right direction is.

Reply:
Oh right, I can take out the generic panel styles from the site-header, but keep the `site-header-panel-links`, `site-header-panel-cards`, and `site-header-panel-featured-content` since these are header-specific

Reply:
Re-opening this thread cause after thinking for a bit I decided to bring back the absolute positioning to the panel. Since panel is not a standalone component, I thought it would make more sense and if needed it can be changed with `position-relative` atomic. Lmk if that works for you.